command/monitor/monitor.go

```diff
@@ -0,0 +1,172 @@
+package monitor
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+)
+
+// Monitor provides a mechanism to stream logs using go-hclog
+// InterceptLogger and SinkAdapter. It allows streaming of logs
+// at a different log level than what is set on the logger.
+type Monitor interface {
+	// Start returns a channel of log messages which are sent
+	// ever time a log message occurs
+	Start() <-chan []byte
+
+	// Stop de-registers the sink from the InterceptLogger
+	// and closes the log channels
+	Stop()
+}
+
+// monitor implements the Monitor interface
+type monitor struct {
+	// protects droppedCount and logCh
+	sync.Mutex
+
+	sink log.SinkAdapter
+
+	// logger is the logger we will be monitoring
+	logger log.InterceptLogger
+
+	// logCh is a buffered chan where we send logs when streaming
+	logCh chan []byte
+
+	// doneCh coordinates the shutdown of logCh
+	doneCh chan struct{}
+
+	// droppedCount is the current count of messages
+	// that were dropped from the logCh buffer.
+	// only access under lock
+	droppedCount int
+	bufSize      int
+	// droppedDuration is the amount of time we should
+	// wait to check for dropped messages. Defaults
+	// to 3 seconds
+	droppedDuration time.Duration
+}
+
+// New creates a new Monitor. Start must be called in order to actually start
+// streaming logs
+func New(buf int, logger log.InterceptLogger, opts *log.LoggerOptions) Monitor {
```

Comment:
Why two constructors, `New` and `new`?  (The latter is a Go keyword so I would encourage changing or eliminating it.)

On a side note, I think there's some disagreement in the Go community about naming of constructors.  On the one hand, some dislike "stuttering" where the same base string is repeated as in `monitor.NewMonitor`, and they would probably prefer `monitor.New`.  Others such as myself would rather have a more descriptive identifier that stands alone of the package name and hence would prefer `NewMonitor`.  This also helps future-proof you in case at some point the monitor package exposes other things that need constructors... it would be weird and inconsistent to have `monitor.New` to create a Monitor but then `monitor.NewOtherThing` to create something else.

Reply:
See my above comment

Reply:
I don't have any strong preferences here re: naming, so I'm happy to change the first to `NewMonitor` and the second to `newMonitor`, I guess?  That remains descriptive and also avoids using a Go keyword.